versioning_function.sql

```diff
@@ -68,7 +68,7 @@ BEGIN
     END IF;
 
     -- check if history table exits
-    IF to_regclass(history_table) IS NULL THEN
+    IF to_regclass(history_table::cstring) IS NULL THEN
```

Comment:
Is there a specific reason for casting history_table to cstring? It appears that the function `to_regclass(cstring)` doesn't exist.

Reply:
I had the opposite problem:

```ERROR:  function to_regclass(text) does not exist at character 8
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
```

Reply:
```sql

SELECT n.nspname as "Schema",
	  p.proname as "Name",
	  pg_catalog.pg_get_function_result(p.oid) as "Result data type",
	  pg_catalog.pg_get_function_arguments(p.oid) as "Argument data types",
	 CASE
	  WHEN p.proisagg THEN 'agg'
	  WHEN p.proiswindow THEN 'window'
	  WHEN p.prorettype = 'pg_catalog.trigger'::pg_catalog.regtype THEN 'trigger'
	  ELSE 'normal'
	 END as "Type"
	FROM pg_catalog.pg_proc p
	     LEFT JOIN pg_catalog.pg_namespace n ON n.oid = p.pronamespace
	WHERE p.proname = 'to_regclass'
	ORDER BY 1, 2, 4;```

Reply:
interesting: docs for 9.6 and 10 state that the parameter should be "text" https://www.postgresql.org/docs/current/static/functions-info.html

Can you try and explicitly cast to text and see what happens?

Reply:
Yeah, that's where I got the error from. Attempting an explicit text cast.

Reply:
This seems to have changed between 9.5 and 9.6:

9.5:
```
   Schema   |    Name     | Result data type | Argument data types |  Type
------------+-------------+------------------+---------------------+--------
 pg_catalog | to_regclass | regclass         | cstring             | normal
```

9.6
```
   Schema   |    Name     | Result data type | Argument data types |  Type
------------+-------------+------------------+---------------------+--------
 pg_catalog | to_regclass | regclass         | text                | normal
```